fix ssh cli defaults (#75245)

bcoca opened the pull request:
* fix ssh cli defaults

  fixes #75221

(cherry picked from commit e5a2fe4c419740e9a709b07b064063f54277b983)

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
connection/ssh

bcoca committed:
fix ssh cli defaults (#75245)

* fix ssh cli defaults

  fixes #75221

(cherry picked from commit e5a2fe4c419740e9a709b07b064063f54277b983)

bcoca committed:
github conflict?

sivel commented on lib/ansible/plugins/connection/ssh.py:
```diff
@@ -167,7 +171,7 @@
           cli:
             - name: ssh_extra_args
-      retries:
+      reconnection_retries:
```
Was this rename purposeful in this backport?  It won't work without changing the `get_option` call.

bcoca committed:
removed retries backrpot

  also removed tests since they rely on features that don't exist in this version
